recipes/full_finetune_single_device.py

```diff
@@ -341,24 +343,31 @@ def train(self) -> None:
                 # Check if this is the norm or not
                 pbar.set_description(f"{curr_epoch+1}|{idx+1}|Loss: {loss.item()}")
                 if self.total_training_steps % self._log_every_n_steps == 0:
+                    log_dict = {
+                        "loss": loss.item(),
+                        "lr": self._optimizer.param_groups[0]["lr"],
+                        "gpu_resources": torch.cuda.memory_allocated(),
+                        "avg_it_s": perf_monitor.get_metric_val("avg_it_s", 0),
```

Comment:
where is perf_monitor defined?

Reply:
probably means this?
```suggestion
                        "avg_it_s": monitor.get_metric_val("avg_it_s", 0),
```